### DISCUSSION TITLE
How to use Cordova plugins without Ionic Native

### DISCUSSION THREAD

**Discussion by D-Marc1** [ionic-team/capacitor#3598] [Q&A] [answered]
I can't find any resources showing how to use Cordova plugins with Capacitor, without Ionic Native. [The official guide](https://capacitorjs.com/docs/cordova/using-cordova-plugins) only shows the following.

> npm install cordova-plugin-name
> npx cap sync

What should the `import` statement look like in the JavaScript?

**Comment by danielehrhardt** [comment]
You could use ionic Native in Capacitor.
If you do not want to use it you could do:
```
declare var cordova: any;
cordova.plugins.plugin.function()
```

**Reply by korinaleo** [reply]
I don't understand.

The old code used:
import { DBMeter } from 'cordova-plugin-dbmeter';
constructor(private dbMeter: DBMeter,

what after ?
declare var cordova: any;

How to use the function?

**Comment by imhoffd** [comment, marked as answer]
There is no `import`, you likely have to access the plugin directly off of `window`. Each plugin's docs may or may not have instructions for direct usage, but you could always inspect the `plugin.xml` for `<js-module>` declarations. For example, in [this plugin](https://github.com/randdusing/cordova-plugin-bluetoothle/blob/master/plugin.xml#L12-L14), it adds the module to `window.bluetoothle`.

**Reply by Anacrusisllc** [reply]
@imhoffd Thanks for providing this tip. I'm in this exact situation but am not getting anywhere. I've inspected the [plugin](https://github.com/manuelnelson/cordova-plugin-midi-sender/blob/0.1.0/plugin.xml) xml file and it tells me:
```
<js-module src="www/midisender.js" name="midisender">
        <clobbers target="MIDISender" />
    </js-module>
```
however when do I this:
`declare var MIDISender: any;`

and then try to access any methods within such as,
`MIDISender.sendProgramChange(1, 30);`
I get: [error] - ERROR Error: Uncaught (in promise): TypeError: undefined is not an object (evaluating 'MIDISender.sendProgramChange').
Whatever path I try window.MIDISender or cordova.plugins.MIDISender I get the same error. Any suggestions or further tips as to how to access the plugin?

**Reply by mlynch** [reply]
That just says to me the plugin didn't install correctly. Did you run `npx cap sync`?

Another way to test this is to connect to your app once it launches through a debugger (`chrome://inspect` on android or Safari Dev Tools on iOS) and see if your plugin is available on `window`.

**Reply by Anacrusisllc** [reply]
@mlynch Thanks! I did run npx cap sync. I created a new project, installed the plugin and used a debugger and found out my plugin was not available. I did the same process with another cordova plugin and was able to get to the plugin through the debugger so I'm guessing my original plugin doesn't work. I will try a different fork and see if that works, otherwise, I guess I'll need to see if I can wrap the Webmidi shim in a capacitor plugin and get it to work. It would be a nice contribution to be able to access a midi keyboard on Ionic iOS projects. There was a request for one which is what brought me to that old plugin in the first place.

**Reply by Anacrusisllc** [reply]
Just to followup, I found an older fork that doesn't support android, but it does work. I noticed in the config.xml file of the old fork the clobbers target was set to cordova.plugins.MIDISender. In the newer version it was just MIDISender. Perhaps that is the mistake depending on what else was changed...
